RedHatInsights/curiosity-frontend
# WIP feat(build): issues/527, docker for local run

## What's included

- feat(build): issues/527, docker for local run
   * docker, leverage existing bundles, local
   * scripts, express serve
   * package.json, scripts

### Notes
- The NPM scripts **are NOT finalized**
- This currently still requires a user to setup the Platform's proxy container and hosts file updates
- The overall process for local compile is slow, really really slow. In order to speed it up we're considering a few alternatives
   - we can pull the bundles from https://github.com/RedHatInsights/curiosity-frontend-build ... since it only releases specific branch we could also leverage this repo as "general artifact storage"... meaning we dump custom feature branch bundles here too
   - another alternative is we link Curiosity and Docker then automatically precompile the images from specific branches, store them in Docker... but this introduces yet another account



## How to test
Testing is subject to change based on updates...

1. Setup the Platform's proxy and hosts file updates
   - You can install from the Platform or attempt to use the baked in NPM script
      - ```$ yarn api:docker```
1. Make sure Docker is running, and you've increased it's resources, we utilized 4 CPUs, 6 GB of Memory
1. Clone these changes down and run 
   - ```$ yarn build:docker2```
   - ```$ yarn start:docker2```
1. Make sure start up the Proxy
   - If you haven't already run ```$ yarn api:docker``` and confirm your browser opened https://ci.foo.redhat.com/subscriptions/rhel-sw/all ... you may have to refresh to see the GUI






## Example

...

## Updates issue/story

#527
